src/main/java/twilightforest/enums/StructureWoodVariant.java

```diff
@@ -32,40 +36,29 @@
     MINE         ( TFBlocks.mine_planks        , TFBlocks.mine_stairs        , TFBlocks.mine_slab         , TFBlocks.mine_button        , TFBlocks.mine_fence        , TFBlocks.mine_gate          , TFBlocks.mine_plate             ),
     SORT         ( TFBlocks.sort_planks        , TFBlocks.sort_stairs        , TFBlocks.sort_slab         , TFBlocks.sort_button        , TFBlocks.sort_fence        , TFBlocks.sort_gate          , TFBlocks.sort_plate             );
 
-    StructureWoodVariant(
-            Block planks,
-            Block stairs,
-            Block slab  ,
-            Block button,
-            Block fence ,
-            Block gate  ,
-            Block plate
-    ) {
-        this.planks     = planks;
-        this.stairs     = stairs;
-        this.slab       = slab  ;
-        this.button     = button;
-        this.fence      = fence ;
-        this.gate       = gate  ;
-        this.plate      = plate ;
+    private final List<Block> blocks;
+
+    StructureWoodVariant(Block ... blocks) {
+        this.planks     = blocks[0]; //;planks;
+        this.stairs     = blocks[1]; // stairs;
+        this.slab       = blocks[2]; // slab  ;
+        this.button     = blocks[3]; // button;
+        this.fence      = blocks[4]; // fence ;
+        this.gate       = blocks[5]; // gate  ;
+        this.plate      = blocks[6]; // plate ;
+        this.blocks     = Arrays.asList(blocks);
     }
 
-    StructureWoodVariant(
-            Supplier<Block> planks,
-            Supplier<Block> stairs,
-            Supplier<Block> slab  ,
-            Supplier<Block> button,
-            Supplier<Block> fence ,
-            Supplier<Block> gate  ,
-            Supplier<Block> plate
-    ) {
-        this.planks     = planks.get();
-        this.stairs     = stairs.get();
-        this.slab       = slab.get()  ;
-        this.button     = button.get();
-        this.fence      = fence.get() ;
-        this.gate       = gate.get()  ;
-        this.plate      = plate.get() ;
+    @SafeVarargs
+    StructureWoodVariant(Supplier<Block> ... blocks) {
+        this.planks     = blocks[0].get(); //;planks;
+        this.stairs     = blocks[1].get(); // stairs;
+        this.slab       = blocks[2].get(); // slab  ;
+        this.button     = blocks[3].get(); // button;
+        this.fence      = blocks[4].get(); // fence ;
+        this.gate       = blocks[5].get(); // gate  ;
+        this.plate      = blocks[6].get(); // plate ;
+        this.blocks     = Stream.of(blocks).map(Supplier::get).collect(Collectors.toList());
     }
```

Comment:
why?
passing them explicitly is cleaner than an array whose contents are completely up to convention
at least with parameters we have the param names.

Reply:
this is true, I'll just explicitly create an array instead

Reply:
Reverted and changed to an explicit array creation.